### DISCUSSION TITLE
Best practice to get rid of reflections: Polarisation filters

### DISCUSSION THREAD

**Discussion by bjoernmayer** [jomjol/AI-on-the-edge-device#1862] [General]
### Issue

One could put a pol filter on the camera and another one on the led.
Now you need to rotate one of them until the reflections are gone. It is a bit of try and error, but it is worth it

What do you think?

**Comment by bucklevision** [comment]
You only need one on the camera, I think, light reflecting off flat surfaces tends to be polarised so you can filter out the reflected light. I've never found a polarising filter did much in single-source artificial light situations - they're super cheap, so easy to try.

**Comment by dominikandreas** [comment]
Have you tried this? I ran into some false detections on my gas meter display (reflections in line-form on rotary digits), e.g. a 7 misclassified as a 2. Currently my solution is mounting the camera at an angle so that reflections of the LED are not on top of the digits. This does distort the image a bit and might have an impact on the precision of the neural net. I wonder if defining corners. Maybe an image transformation could help with this given 4 identifiable corner points around the area to digitize, but wondering if polarization filter would also suffice (easier to implement).

**Reply by bjoernmayer** [reply]
Yep.
I reused some old 3D-Glasses. I've put one filter over the lense. The other one I didn't attach yet.
I then installed the Example ESP32 Camera Server on it and turned on led flash.
I pointed the camera at something reflective and kept turning the pol filter. At one point reflections are nearly gone.

**Comment by JanneN9** [comment]
I installed ESP32-board at a 10-degree angle (led) (camera direct to lcd) and a small "diffuse" front of the led. It is working, but adjusting that combination takes days.
![Diffuse](https://user-images.githubusercontent.com/10402422/216836342-2563f799-7bc8-4e4b-a37d-b6f0a3bd42dc.jpg)
